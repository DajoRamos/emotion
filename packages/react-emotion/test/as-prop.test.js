import 'test-utils/legacy-env'
import React from 'react'
import * as renderer from 'test-utils/compat-render-json'
import styled from 'react-emotion'

describe('`as` prop', () => {
  test('string base - string as', () => {
    const Title = styled('h1')`
      color: green;
    `

    const tree = renderer
      .create(
        <article>
          <Title>My Title</Title>
          <Title as="h2">My Subtitle</Title>
        </article>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('styled string base - string as', () => {
    const Title = styled('h1')`
      color: green;
    `

    const Subtitle = styled(Title)`
      font-weight: 100;
    `

    const tree = renderer
      .create(
        <article>
          <Title>My Title</Title>
          <Subtitle>My Subtitle</Subtitle>
          <Subtitle as="h3">My smaller Subtitle</Subtitle>
        </article>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('composite base - string as', () => {
    const Title = styled(props => <h1 {...props} />)`
      color: green;
    `

    const tree = renderer
      .create(
        <article>
          <Title>My Title</Title>
          <Title as="h3">My Subtitle</Title>
        </article>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('forward as - string base', () => {
    const Title = styled('h1', {
      shouldForwardProp: prop => prop !== 'theme'
    })`
      color: green;
    `

    const tree = renderer
      .create(
        <article>
          <Title>My Title</Title>
          <Title as="h2">My Subtitle</Title>
        </article>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('forward as - composite base', () => {
    const Title = styled(props => <h1 {...props} />, {
      shouldForwardProp: prop => prop !== 'theme'
    })`
      color: green;
    `

    const tree = renderer
      .create(
        <article>
          <Title>My Title</Title>
          <Title as="h2">My Subtitle</Title>
        </article>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
