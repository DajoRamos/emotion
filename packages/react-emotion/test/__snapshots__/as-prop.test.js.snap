// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`as\` prop composite base - string as 1`] = `
.emotion-0 {
  color: green;
}

<article>
  <h1
    className="emotion-0 emotion-1"
  >
    My Title
  </h1>
  <h3
    className="emotion-0 emotion-1"
  >
    My Subtitle
  </h3>
</article>
`;

exports[`\`as\` prop forward as - composite base 1`] = `
.emotion-0 {
  color: green;
}

<article>
  <h1
    className="emotion-0 emotion-1"
  >
    My Title
  </h1>
  <h1
    as="h2"
    className="emotion-0 emotion-1"
  >
    My Subtitle
  </h1>
</article>
`;

exports[`\`as\` prop forward as - string base 1`] = `
.emotion-0 {
  color: green;
}

<article>
  <h1
    className="emotion-0 emotion-1"
  >
    My Title
  </h1>
  <h1
    as="h2"
    className="emotion-0 emotion-1"
  >
    My Subtitle
  </h1>
</article>
`;

exports[`\`as\` prop string base - string as 1`] = `
.emotion-0 {
  color: green;
}

<article>
  <h1
    className="emotion-0 emotion-1"
  >
    My Title
  </h1>
  <h2
    className="emotion-0 emotion-1"
  >
    My Subtitle
  </h2>
</article>
`;

exports[`\`as\` prop styled string base - string as 1`] = `
.emotion-0 {
  color: green;
}

.emotion-2 {
  color: green;
  font-weight: 100;
}

<article>
  <h1
    className="emotion-0 emotion-1"
  >
    My Title
  </h1>
  <h1
    className="emotion-2 emotion-3"
  >
    My Subtitle
  </h1>
  <h3
    className="emotion-2 emotion-3"
  >
    My smaller Subtitle
  </h3>
</article>
`;
